src/librustc_mir/build/matches/mod.rs

```diff
@@ -534,12 +535,12 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
                     self.visit_bindings(subpattern, pattern_user_ty, f);
                 }
             }
+
             PatternKind::Array {
                 ref prefix,
                 ref slice,
                 ref suffix,
-            }
-            | PatternKind::Slice {
+            } | PatternKind::Slice {
```

Comment:
This preferred or is the following preferred?

```rust
Foo {
   bar: 1
} |
Foo {
  bar: 2
}
```

This could also probably be formatted like so

```rust
PatternKind::Array { ref prefix, ref slice, ref suffix } |
PatternKind::Slice { ref prefix, ref slice, ref suffix
```

Reply:
How about the following? It seems to be most readable and agree with existing styles.

```
Foo {
   bar: 1
}
| Foo {
  bar: 2
}
```